### DISCUSSION TITLE
Support pre-compiled ROOT binaries and installations from package managers

### DISCUSSION THREAD

**Discussion by eguiraud** [CheckMATE2/checkmate2#8] [Ideas]
Hello,
ROOT dev here. In the [installation instructions](https://checkmate.hepforge.org/tutorial/ver2/7_install_root.php?pythonpath=%2Fhome%2F&rootpath=%2Fhome%2Froot) it is mentioned that, for usage with CheckMATE2, ROOT should be compiled from source and pre-compiled binaries are not supported.

Compiling ROOT from source is often a source of confusion or problems for many ROOT users (see e.g. [the latest, checkMATE-related post on the ROOT forum](https://root-forum.cern.ch/t/root-6-22-no-python-support-enabled/44131) about issues compiling ROOT from source) and the ROOT team is trying to push for users to install via package managers (preferably) or from pre-compiled binaries -- see also our recently-refurbished installation page: https://root.cern/install .

We would be glad to help figuring out the linkage issues mention in the CheckMATE2 installation instructions so that usage of a pre-compiled or packaged ROOT is supported!

Cheers,
Enrico

**Comment by NishitaDesai** [comment]
Dear Enrico,

Thanks for your message.  Yes, it would be good if precompiled ROOT works out of the box for CheckMATE.  We sort of started advising against it when we were using root5 and wanted some statistical machinery that used to not be included by default.  But with root6, things seem to be much simpler. 

We haven’t redone the installation worksheet from scratch.  We’re sort of having trouble with having enough person power to test everything at the moment, but we will get back to you once we have someone test the precompiled versions out on Linux and Mac platforms.

Best,
Nishita

**Comment by krolbiecki** [comment]
Hi Enrico,

as far as I can tell the pre-compiled root for Ubuntu works out of the box with CheckMATE. Thanks for pointing out the outdated instructions on the website. We will attend to it soon.

Cheers,
Krzysztof

**Reply by eguiraud** [reply]
Great!

**Reply by eguiraud** [reply]
If special CheckMATE-related ROOT tweaks are not needed anymore, you could point to https://root.cern/install to "outsource" the ROOT installation instructions if you want.

**Comment by agagsgroove** [comment]
Hi all,

I have slighty changed the installation details of CheckMATE and we refer to the official root install webpage.

Cheers,

Jong

**Reply by eguiraud** [reply]
Hi @agagsgroove ,
I quickly went through CheckMATE's installation instructions and there are still a number of inaccuracies:

- [here](https://checkmate.hepforge.org/tutorial/ver2/7_install_root.php?pythonpath=%2Fhome%2F&rootpath=%2Fhome%2Froot): usage of configure/make is suggested, but ROOT moved to use cmake exclusively years ago. Also the page points to http://root.cern.ch/drupal/content/downloading-root but the correct address is now https://root.cern/install (which the wrong link currently redirects to)
- [here](https://checkmate.hepforge.org/tutorial/check_root.php?python=local) a distinction is made between installations of ROOT at `/usr/bin/root` vs other locations. In general, package managers might install ROOT also in `/usr/local/bin` or other locations based on the Linux distro. As I believe the intention is to distinguish between ROOT installed via package manager and ROOT built from source, checking whether ROOT is installed at `/usr/local/bin` or not is not a good proxy 
- [here](https://checkmate.hepforge.org/tutorial/check_root2.php?root=global&python=local) (and maybe elsewhere) the command `root-config --has-python` is used to check for Python support. In recent ROOT versions the correct flag is `--has-pyroot` and `--has-python` simply prints `no` (this backward-incompatible change in ROOT was actually what confused the CheckMATE user that contacted us on the forum, see my first post). I [opened an issue](https://github.com/root-project/root/issues/7662) about this terrible behavior of `root-config`, but in the meanwhile you might want to change the instructions to check that `root-config --features` contains one of `pyroot` or `python`
- [here](https://checkmate.hepforge.org/tutorial/check_root3.php?root=global&python=local): adding packages to an existing ROOT build is tricky/unsupported. The [instructions to do so](https://checkmate.hepforge.org/tutorial/install_root_packages.php?root=global&python=local) suggest usage of configure/make instead of CMake (i.e. they are outdated, would not work anymore) and anyway overwriting a package manager installation with a manual `make install` is a terrible practice: it is bound to cause a number of issues that are hard to foresee and hard to debug

Sorry for being pedantic, but we often see non-expert users getting confused by these things. If I may suggest a simplification to the tutorial: when it comes to installing ROOT, you could just point users to https://root.cern/install and tell them to make sure that `root-config --features` contains `roofit`, `minuit2` and `pyroot` or `python` when they are done. All pre-compiled ROOT binaries and all system packages should satisfy these requirements.
